cns/cnsclient/cnsclient_test.go

```diff
@@ -0,0 +1,142 @@
+package cnsclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/common"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+	"github.com/Azure/azure-container-networking/log"
+)
+
+var (
+	testNCID = "06867cf3-332d-409d-8819-ed70d2c116b0"
+
+	testIP1      = "10.0.0.1"
+	testPod1GUID = "898fb8f1-f93e-4c96-9c31-6b89098949a3"
+	testPod1Info = cns.KubernetesPodInfo{
+		PodName:      "testpod1",
+		PodNamespace: "testpod1namespace",
+	}
+)
+
+func addTestStateToRestServer(svc *restserver.HTTPRestService) {
+	// set state as already allocated
+	state1, _ := restserver.NewPodStateWithOrchestratorContext(testIP1, 24, testPod1GUID, testNCID, cns.Available, testPod1Info)
+	ipconfigs := []cns.ContainerIPConfigState{
+		state1,
+	}
+	svc.AddIPConfigsToState(ipconfigs)
+}
+
+func TestMain(m *testing.M) {
+	tmpFileState, err := ioutil.TempFile(os.TempDir(), "cns-*.json")
+	tmpLogDir, err := ioutil.TempDir("", "cns-")
+	fmt.Printf("logdir: %+v", tmpLogDir)
+
+	if err != nil {
+		panic(err)
+	}
+
+	defer os.RemoveAll(tmpLogDir)
+	defer os.Remove(tmpFileState.Name())
+
+	if err != nil {
+		panic(err)
+	}
+
+	logger.InitLogger("azure-cns.log", 0, 0, tmpLogDir)
+	config := common.ServiceConfig{}
+
+	httpRestService, err := restserver.NewHTTPRestService(&config)
+	svc := httpRestService.(*restserver.HTTPRestService)
+	svc.Name = "cns-test-server"
+	if err != nil {
+		logger.Errorf("Failed to create CNS object, err:%v.\n", err)
+		return
+	}
+
+	addTestStateToRestServer(svc)
+
+	if httpRestService != nil {
```

Comment:
Would we want to continue if httpRestService is nil?

Reply:
this is just test main, should be fine